### PR TITLE
Simplify argument parsing

### PR DESCRIPTION
**Pull request by lnicola**
and use `Display` when printing errors.

### CODE REVIEW

File: crates/rust-analyzer/src/bin/main.rs [hunk 1]
```diff
@@ -131,3 +127,10 @@ fn run_server() -> Result<()> {
     log::info!("... IO is down");
     Ok(())
 }
+
+fn main() {
+    if let Err(e) = run() {
+        eprintln!("{}", e);
+        std::process::exit(1);
+    }
+}
```
Comment (matklad) [inline crates/rust-analyzer/src/bin/main.rs:136]: Let's use these naming, order & exit code: https://github.com/rust-analyzer/ungrammar/blob/da1aa8a99738c1a4fe5e321e39aae857c1eb54c8/ungrammar2json/src/main.rs#L9-L16

File: crates/rust-analyzer/src/bin/args.rs [hunk 2]
```diff
@@ -74,10 +74,11 @@ impl Args {
...
                     print_subcommands();
-                    return Ok(Args { verbosity, command: Command::Help });
+                    String::from("help")
+                } else {
+                    matches.finish().or_else(handle_extra_flags)?;
+                    return Ok(Args { verbosity, command: Command::RunServer });
                 }
-                matches.finish().or_else(handle_extra_flags)?;
-                return Ok(Args { verbosity, command: Command::RunServer });
             }
```
Comment (matklad) [inline crates/rust-analyzer/src/bin/args.rs:82]: The early return versions reads better to me, and it also seems less verbose. Duplicating `verbosity` in unfortuante, but berpapse we can just `let help = ...; return help`?

Comment (lnicola) [reply crates/rust-analyzer/src/bin/args.rs:82]: Done

Comment (matklad) [reply crates/rust-analyzer/src/bin/args.rs:82]: bors r+

**Pull request merged**